packages/wrangler/src/d1/migrations/helpers.ts

```diff
@@ -124,16 +124,14 @@ function getMigrationNames(migrationsPath: string): Array<string> {
 	return migrations;
 }
 
+/**
+ * Returns the highest current migration number plus one, ignoring any missing numbers.
+ */
 export function getNextMigrationNumber(migrationsPath: string): number {
-	let highestMigrationNumber = -1;
-
-	for (const migration in getMigrationNames(migrationsPath)) {
-		const migrationNumber = parseInt(migration.split("_")[0]);
-
-		if (migrationNumber > highestMigrationNumber) {
-			highestMigrationNumber = migrationNumber;
-		}
-	}
+	const migrationNumbers = getMigrationNames(migrationsPath).map((migration) =>
+		parseInt(migration.split("_")[0])
+	);
+	const highestMigrationNumber = Math.max(...migrationNumbers, 0);
```

Comment:
This explicitly makes the first number 0 + 1, which is a regression to how it works currently where the first number is 0.

If regression is unwanted, putting it to `-1` will make it -1 + 1 = 0.
```suggestion
	const highestMigrationNumber = Math.max(...migrationNumbers, -1);
```